disco/extensions/upgrade_simulation/upgrades/common_functions.py

```diff
@@ -1128,7 +1138,69 @@ def get_thermal_equipment_info(compute_loading, equipment_type, upper_limit=None
     else:
         raise Exception(f"Undefined multiplier_type {multiplier_type} passed.")   
     return loading_df
+
+
+def apply_timeseries_step(equipment_type, **kwargs):
+    """
+    Time-series data used to determine thermal and voltage violations
+    """
+    if (equipment_type.lower() == "transformer") or (equipment_type.lower() == "line"):
+        deciding_column_name = kwargs.get("deciding_column_name")
+        compute_loading = kwargs.get("compute_loading")
+        upper_limit = kwargs.get("upper_limit")
+        ignore_switch = kwargs.get("ignore_switch", False)
+    elif equipment_type == "voltage":  # pop out of dictionary since kwargs are also passed to voltage function
+        voltage_upper_limit = kwargs.pop("voltage_upper_limit")
+        voltage_lower_limit = kwargs.pop("voltage_lower_limit")
+        raise_exception = kwargs.pop("raise_exception")
+        deciding_column_dict = kwargs.pop("deciding_column_dict")
+    else:
+        raise Exception(f"Incorrect equipment type {equipment_type} provided. Possible values: line, transformer, voltage.")
+        
+    comparison_dict = {}
+    sol = dss.Solution  # opendss solution interface
+    dss.run_command("Set mode=yearly number=1")  # 1 power flow per solve and I guess stepsize remains the same

```

Comment:
"I guess" is not a great comment in code. The response from the next developer will be, "so you're making me look this up to be confident that it's true?"